test/module/iroha-cli/client_test.cpp

```diff
@@ -45,6 +45,7 @@ using namespace shared_model::proto;
 using namespace std::chrono_literals;
 constexpr std::chrono::milliseconds initial_timeout = 1s;
 constexpr std::chrono::milliseconds nonfinal_timeout = 2 * 10s;
+constexpr unsigned status_read_attempts = 3;
```

Comment:
Why are 3 attempts enough? Seems like we improve the probability of test passing instead of doing it deterministic.

Reply:
The deterministic way to do it is to access the status bus and do additional synchronization. But the idea from discussion was to replicate the conditions of the client which tries to use the API.

Reply:
Here we imitate the behavior of StatusStram client but on a bit lower level. So the do-while cycle imitates client resubscription to the stream. Stream "expiration" is a valid designed case (see #1615 for the details).

The number of attempts (3) is a magic constant here. The idea behind this number is the following: only one resubscription is usually enough to pass the test; if three resubscribes were not enough, then most likely threre is another bug.

Reply:
Discussed offline with @muratovv: decided to leave used approach.